pkg/pubsub/system.go

```diff
@@ -0,0 +1,89 @@
+package pubsub
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/connection"
+	prvd "github.com/open-policy-agent/gatekeeper/pkg/pubsub/provider"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("pubsub-system")
+
+type System struct {
+	mux         sync.RWMutex
+	connections map[string]connection.Connection
+	providers   map[string]string
+}
+
+func NewSystem() *System {
+	return &System{}
+}
+
+func (s *System) Publish(ctx context.Context, connection string, topic string, msg interface{}) error {
+	s.mux.RLock()
+	defer s.mux.RUnlock()
+	if len(s.connections) > 0 {
+		if c, ok := s.connections[connection]; ok {
+			return c.Publish(context.Background(), msg, topic)
+		}
+		log.Info("Connection is not initialized", "name", connection)
+	} else {
+		log.Info("No pub sub tools are enabled")
+	}
+	return nil
+}
+
+func (s *System) UpsertConnection(ctx context.Context, config interface{}, name string, provider string) error {
```

Comment:
can we add unit tests for this? can be a follow up

Reply:
Follow-up will be better to keep the changes small.